### ISSUE TITLE
Plugin > Browser Bookmarks > Open bookmark in origin browser vs system default...and a larger conversation around patterns to define file=>program associations in Windows

### ISSUE THREAD

**Issue by jmtornetta** [Flow-Launcher/Flow.Launcher#2805]
# TLDR
Would love it if Flow could allow me to open a browser bookmark with the "origin" browser (Chrome) instead of my default browser (Brave).

# Q/A for Feature request

**Is your feature request related to a problem? Please describe.**
Kinda, sorta. But the problem is Windows not Flow :)

**Describe the solution you'd like**

Browser favorites are opened with the browser of origin versus default system browser.

**Describe alternatives you've considered**

1. i've written a custom powershell script for this purpose to hijack Windows program launch logic, but it's not w/o problems (see below)
2. looked through Flow plugin repo, didn't see anything
3. asked ChatGPT :). said a custom script needs to be written w/ manual file associations to said script in windows :/

# Additional context
Flow makes searching and launching easier. The searching MUCH easier. But the launching is *somewhat* at the mercy of the OS's default file/program association.

Doesn't seem like Windows has a great way to easily define logic to open different files/links/folders with different programs, based on patterns. It's possible with a batch script but that involve some OS hijacking like association a whole bunch of file types with this script.

However, this opens up a larger question about if Flow could define a simple "pattern" => "program" association table or file to allow users to override Windows default program launch when there's a pattern (i.e. regexp) match.

Examples:

1. would love to open google drive docs with organization chrome, instead of brave
2. open all .ts files in my personal folders with VS Code and others with VS Studio
3. open video files in a "watch" folder with a player (VLC) and those in an "edit" folder with a video editor.

**Comment by systemmonkey42** [comment]
I came here for a similar reason.

The chrome bookmark loader will find all bookmarks across all profiles.
- I would prefer to allow only specific profiles (or only the default)

When a bookmark is matched to a non-default profile, it is currently launched with the default browser
meaning saved cookies/credentials will not work.
- Ensure chrome is launching using the correct profile before launching the bookmark. (equivalent of chrome.exe --profile-directory="{profile}" "{bookmark}")

Obviously, if the bookmark is found in EDGE, don't launch chrome.

**Comment by oo-rbiiix** [comment]
This will be awesome feature. Really needed. Why? 

My default browser is edge, due to m365 and company policies (many bookmarks in there) but in flow launcher cannot be setup my own browser as default, because i need to open my edge bookmarks in edge. But for everything else i need my favourite opera browser. Please...

**Comment by jjw24** [comment]
Please thumbs up on issue description if you would like to see this feature implemented, or comment if you would like to provide additional context.

**Comment by sergebat** [comment]
I want to share that I have created a separate plugin that supports this feature: https://github.com/sergebat/Flow.Launcher.Plugin.MultiprofileBookmarks

Mostly as a proof of concept, but this is robust enough at least for my daily use! The PR in Flow.Launcher.PluginsManifest is not yet approved, so it is not officially available. 

It is Chrome only at this time and does not support favicons. 

I'll be happy if one day this feature is available in the core bookmarks plugin!
